src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -450,6 +450,9 @@ void program::set_options() {
     if (!options.get<build_option_type::force_implementations>()->forcing.empty()) {
         options.set_option(build_option::optimize_data(true));
     }
+
+    // Set device info for subgroup_local_block_io
+    get_engine().get_device()->is_local_block_io_supported();
```

Comment:
Why it needs to be a part of device? I think this check can be a method of program and then you'll just have something like this in kernel selector helper:
```
params.engineInfo.bLocalBlockIOSupport = device_info.supports_local_block_io && prog.is_local_block_io_supported();
```

And I wonder, if we keep the impl in program, then can kernels_cache be reused for compilation/caching? I think this dummy kernel can be wrapped into `kernel_string` structure and then you can do set_kernel_source(); build_all(); reset();

Reply:
The function 'is_local_block_io_supported' had been originally located in creation of ocl_device class.
It was modified to on-demand, because it could not get the cache_dir function in creation.

This function should be used with device info together to check bLocalBlockIOSupport.
So I just left it in ocl_device.

About reusing, Do you mean modify is_local_block_io_support function to use it both kernels_cache and kernel selector helper?

Reply:
If it is located only in kernel selector helper. It is called by compile_graph and executed by multiple threads.
It  requires wider mutex. So I added this line.

About kernels_cache, I will check to use them.

Reply:
Modified.